temporian/io/test/polars_test.py

```diff
@@ -0,0 +1,496 @@
+
+    def test_invalid_string_timestamp_type(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "timestamp": ["A", "B", 1.0],
+                "costs": [740.0, 508.0, 573.0],
+            }
+        )
+
+        with self.assertRaises(ValueError):
+            from_polars(df, indexes=["product_id"], timestamps="timestamp")
+
+    def test_timestamps_with_missing_values(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016, 574016],
+                "timestamp": [1.0, 2.0, 3.0, None],
+                "costs": [740.0, 508.0, 573.0, 573.0],
+            }
+        )
+
+        with self.assertRaises(ValueError):
+            from_polars(df, indexes=["product_id"], timestamps="timestamp")
+
+    def test_timestamps_with_non_supported_object(self) -> None:
+        with self.assertRaises(TypeError):
+            df = pl.DataFrame(
+                {
+                    "product_id": [666964, 666964, 574016, 574016],
+                    "timestamp": [1.0, 2.0, 3.0, object()],
+                    "costs": [740.0, 508.0, 573.0, 573.0],
+                }
+            )
+
+    def test_no_index(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "timestamp": [1.0, 2.0, 3.0],
+                "costs": [740.0, 508.0, 573.0],
+            }
+        )
+        expected_evset = event_set(
+            timestamps=[1.0, 2.0, 3.0],
+            features={
+                "product_id": [666964, 666964, 574016],
+                "costs": [740.0, 508.0, 573.0],
+            },
+        )
+        evset = from_polars(df, indexes=[], timestamps="timestamp")
+        self.assertEqual(evset, expected_evset)
+
+    def test_datetime_in_feature_column(self) -> None:
+        df = pl.DataFrame(
+            {
+                "product_id": [666964, 666964, 574016],
+                "costs": ["2022-01-01", "2022-01-02", "2022-01-03"],
+                "timestamp": [740.0, 508.0, 573.0],
+            }
+        )
+        expected_evset = event_set(
+            timestamps=[740.0, 508.0, 573.0],
+            features={
+                "product_id": [666964, 666964, 574016],
+                "costs": ["2022-01-01", "2022-01-02", "2022-01-03"],
+            },
+        )
+        evset = from_polars(df, indexes=[], timestamps="timestamp")
+        self.assertEqual(evset, expected_evset)
+
+    def test_nan_in_string(self) -> None:
+        evset = from_polars(
+            pl.DataFrame(
+                {
+                    "timestamp": [1, 2, 3],
+                    "x": ["a", math.nan, "b"],
+                    "y": [1, 2, 3],
+                }
+            )
+        )
+        expected_evset = event_set(
+            timestamps=[1, 2, 3],
+            features={
+                "x": ["a", "None", "b"],
```

Comment:
any idea why this yields "None" but pandas' test yields "nan"?

Reply:
this is because "NaN values are considered to be a type of floating point data and are not considered to be missing data in Polars"

Ref:<https://docs.pola.rs/user-guide/expressions/missing-data/#notanumber-or-nan-values>